### PR TITLE
Lunarfusion/css 122 badge migration to core tokens

### PR DESCRIPTION
**Pull request by lunarfusion**
## Description
BREAKING CHANGE: This migrates the Badge component to core tokens.
- Jira issue CSS-122 https://jira.corp.adobe.com/browse/CSS-122
- Note that icon spacing impacts the alignment of the badge

## How and where has this been tested?
 - Chrome 103 for macOS
 - Firefox 102 for macOS
 - Safari 15.5 for macOS
 - **How this was tested:** comparing local build of http://localhost:3000/docs/badge.html with XD redline components batch 2 and with https://spectrum.adobe.com/page/badge/

## Screenshots

Screen Shot 2022-08-08 at 12 19 30 PM
Screen Shot 2022-08-10 at 8 35 21 AM
Screen Shot 2022-08-08 at 12 19 42 PM
Screen Shot 2022-08-08 at 12 19 48 PM
Screen Shot 2022-08-08 at 12 19 55 PM

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [ ] I have tested these changes in Windows High Contrast mode.

- [ ] This pull request is ready to merge.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,5 +14,7 @@ npm-debug.log*
 /backstop_data/ci_report/
 /backstop_data/backstop_scenarios.json
 /backstop_data/packageDependentMap.json
+backstop.json
```
Comment (bernhard-adobe) [inline .gitignore:17]: I think we should still remove those from our gitignore files like we chatted in an earlier discussion.

File: components/badge/index.css [hunk 2]
```diff
@@ -10,59 +10,279 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+    background: var(--mod-badge-background-color-negative,
+                    var(--spectrum-badge-background-color-negative));
+  }
+
+  &--positive {
+    background: var(--mod-badge-background-color-positive,
+                    var(--spectrum-badge-background-color-positive));
+  }
+
+  /* fixed position variants with border radius 0 on the fixed edge of the component */
```
Comment (bernhard-adobe) [inline components/badge/index.css:190]: Excellent!

**Review by bernhard-adobe — APPROVED** [review]
Approved, great work!!!

**Comment by bernhard-adobe** [comment]
VRT test is: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/28/

File: .gitignore (hunk 1 above)
Comment (lunarfusion) [reply .gitignore]: New commit resolves this. https://github.com/adobe/spectrum-css/commit/70145a937c2c0239fcea28987a1054856aef0249

**Comment by lunarfusion** [comment]
Note that a new commit https://github.com/adobe/spectrum-css/commit/750c4f4fd3382166a2cc1812044d42106b17c3b7 resolves the issue of consistent vertical alignment of badge components. Badges will now center vertically so that badges with and without icons will align with each other.

**Comment by bernhard-adobe** [comment]
VRT test is 	https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/33/artifact/backstop_data/html_report/index.html
[2022-08-10 -- 1494  - badge.zip](https://github.com/adobe/spectrum-css/files/9306241/2022-08-10.--.1494.-.badge.zip)

**Review by pfulton — APPROVED** [review]

**Pull request merged**